vaadin-text-field-flow-parent/vaadin-text-field-flow/src/test/java/com/vaadin/flow/component/textfield/tests/IntegerFieldTest.java

```diff
@@ -59,9 +66,39 @@ public void assertStepGreaterThanZero() {
 
     @Override
     @Test
-    public void initialValuePropertyValue() {
-        assertEquals(field.getEmptyValue(),
-                field.getElement().getProperty("value"));
+    public void initialValueIsNotSpecified_valuePropertyHasEmptyString() {
+        IntegerField integerField = new IntegerField();
+        Assert.assertNull(integerField.getValue());
+        Assert.assertEquals("", integerField.getElement().getProperty("value"));
+    }
+
+    @Override
+    @Test
+    public void initialValueIsNull_valuePropertyHasEmptyString() {
```

Comment:
**note:** NumberField's constructor doesn't have overloads that accept an initial value.